src/components/organization/OrganizationsPartnerList.js

```diff
@@ -0,0 +1,154 @@
+
+        if(id !== undefined){
+            usersAPI.getHrPartnersByOrganization(id)
+                .then(data => updateUsers(data))
+        }
+    }, [id])
+
+    const defaultFormState = {
+        firstName: "",
+        lastName: "",
+        mail: ""
+    }
+
+
+    const users = props.users || []
+    const {handleSubmit, control, reset} = useForm({defaultValues:defaultFormState})
+
+    const onUserAdd = (data) => {
+        withSwal({
+            loadingTitle: "Adding HR user",
+            promise: () => props.onSubmit(data),
+            successSwalTitle: "Success",
+            successSwalText: "You've successfully added HR user!",
+            successFunction: () => {
+                props.onChange(users.concat([data]))
+                reset()
+            },
+            errorSwalTitle: "We couldn't save this user for you"
+        })
+        if(props.onSubmit){
+            props.onSubmit(data)
+        }
+    }
+
+    const onUserDelete = (idx, data) => {
+        withSwal({
+            loadingTitle: "Deleting HR user",
+            promise: () => props.onDelete(data),
+            successSwalTitle: "Success",
+            successSwalText: "You've successfully deleted HR user!",
+            successFunction: () => {
+                props.onChange(users.slice(0, idx).concat(users.slice(idx+1, undefined)))
+                reset()
+            },
+            errorSwalTitle: "We couldn't delete this user for you"
+        })
+        if (props.onDelete) {
+            props.onDelete(data)
+        }
+    }
+
+    return <>
+        <form id="user-form" name="user-form" onSubmit={handleSubmit(onUserAdd)}></form>
+        <Grid item xs={12} sm={6}>
+            <Controller
+                control={control}
+                name="firstName"
+                rules={{required: true}}
+                defaultValue={defaultFormState.firstName}
+                render={({field}) =>
+                    <TextField
+                        {...field}
+                        label="First Name"
+                        variant="outlined"
+                        form="user-form"
+                        fullWidth />
+                }
+            />
+        </Grid>
+        <Grid item xs={12} sm={6}>
+            <Controller
+                control={control}
+                name="lastName"
+                rules={{required: true}}
+                defaultValue={defaultFormState.lastName}
+                render={({field}) =>
+                    <TextField
+                        {...field}
+                        label="First Name"
+                        variant="outlined"
+                        form="user-form"
+                        fullWidth />
+                }
+            />
+        </Grid>
+        <Grid item xs={12} sm={6}>
+            <Controller
+                control={control}
+                name="mail"
+                rules={{required: true, validate: (mail) => !users.some(u => u.mail === mail)}}
```

Comment:
Plus dodałbym jakąś walidację na adres email. Jeśli nie ma jej jeszcze w utilsach, to możesz ją do utilsów wynieść.